spec/asyncapi.md

```diff
@@ -138,6 +138,8 @@ In order to preserve the ability to round-trip between YAML and JSON formats, YA
 An AsyncAPI document MAY be made up of a single document or be divided into multiple,
 connected parts at the discretion of the author. In the latter case, [Reference Objects](#referenceObject) are used.
 
+It is important to note that everything that is defined in an AsyncAPI document MUST be used by the implemented [Application](#definitionsApplication), with the exception of the [Components Object](#componentsObject). Everything that is defined inside the Components Object represents a resource that MAY or MAY NOT be used in the rest of the document and MAY or MAY NOT be used by the implemented [Application](#definitionsApplication).
```

Comment:
Not super happy with this description. I'm strongly accepting suggestions to improve it.

Also, not sure if this belongs here but I think it makes sense to put it under the File Structure section.

Reply:
I see what you mean now. It's actually not necessary to have that last part 👍